### PR TITLE
Undistort 2D points function

### PR DESCRIPTION
**Pull request by jhacsonmeza**
### Description
`undistort_points` function to compensate for lens distortion a set of 2D points. Radial, tangential, thin prism, and tilt distortion are considered. Additionally `inverseTiltProjection` function is used for the tilt distortion compensation.

### CODE REVIEW

**Comment by edgarriba** [comment]
@jhacsonmeza could you add some uni tests? Check the ones e.g. in geometry epipolar

**Comment by jhacsonmeza** [comment]
I've added a unit test `test_opencv` where I used two distortion coefficients to be compared with the results given by OpenCV.

**Comment by edgarriba** [comment]
@jhacsonmeza @dkoguciuk as for the format we don’t have to worry that much any more since we have not the pre-commit bot (update your pr with master for that). As for gradcheck, make sure that any operations doesn’t contain possible zero divisions. Besides, I’d suggest as discussed in slack to provide batched functionality to make easier to compute e.g loss functions

**Comment by jhacsonmeza** [comment]
@dkoguciuk thank you. I'm going to modify the code to be aligned with pep8. I've used pytest to check my unit test, and I'm working in a problem.

**Comment by edgarriba** [comment]
@jhacsonmeza code format shouldn’t be a problem any more since the precommit-ci will fix fir you every time you push a commit to the branch

File: test/geometry/calibration/test_undistort.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        K = torch.rand(B, 3, 3, device=device, dtype=dtype)
+        distCoeff = torch.rand(B, Ndist, device=device, dtype=dtype)
+
+        pointsu = undistort_points(points, K, distCoeff)
+        assert points.shape == (B, N, 2)
+
+    def test_opencv(self, device, dtype):
+        # ----- Test 1: using 5 distortion coefficients
+        pts = torch.tensor(
+            [[1028.0374, 788.7520], [1025.1218, 716.8726], [1022.1792, 645.1857]], device=device, dtype=dtype
```
Comment (edgarriba) [inline test/geometry/calibration/test_undistort.py:39]:
```suggestion
            [[[1028.0374, 788.7520], [1025.1218, 716.8726], [1022.1792, 645.1857]]], device=device, dtype=dtype
```

File: test/geometry/calibration/test_undistort.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            dtype=dtype,
+        )
+        dist = torch.tensor([[-0.1007, 0.2650, -0.0018, 0.0007, -0.2597]], device=device, dtype=dtype)
+
+        # Expected ouput generated with OpenCV:
+        # import cv2
+        # ptsu_expected = cv2.undistortPoints(pts.numpy().reshape(-1,1,2), K.numpy(),
+        #                               dist1.numpy(), None, None, K.numpy()).reshape(-1,2)
+        ptsu_expected = torch.tensor(
+            [[1030.5992, 790.65533], [1027.3059, 718.10020], [1024.0700, 645.90600]], device=device, dtype=dtype
```
Comment (edgarriba) [inline test/geometry/calibration/test_undistort.py:53]:
```suggestion
            [[[1030.5992, 790.65533], [1027.3059, 718.10020], [1024.0700, 645.90600]]], device=device, dtype=dtype
```

File: test/geometry/calibration/test_undistort.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+        points = torch.rand(B, N, 2, device=device, dtype=dtype)
+        K = torch.rand(B, 3, 3, device=device, dtype=dtype)
+        distCoeff = torch.rand(B, Ndist, device=device, dtype=dtype)
+
+        pointsu = undistort_points(points, K, distCoeff)
+        assert points.shape == (B, N, 2)
+
+    def test_opencv(self, device, dtype):
+        # ----- Test 1: using 5 distortion coefficients
```
Comment (edgarriba) [inline test/geometry/calibration/test_undistort.py:37]: Also Split in three different tests
test_opencv_five_coeff
test_opencv_all_coeff
test_opencv_stereo

**Review by edgarriba — CHANGES_REQUESTED** [review]

**Review by edgarriba — APPROVED** [review]

**Review by lferraz — APPROVED** [review]

File: kornia/geometry/calibration/undistort.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    # This is equivalent to K^-1 [u,v,1]^T
+    x: torch.Tensor = (points[..., 0] - cx) / fx  # (BxN - Bx1)/Bx1 -> BxN
+    y: torch.Tensor = (points[..., 1] - cy) / fy  # (BxN - Bx1)/Bx1 -> BxN
+
+    # Compensate for tilt distortion
+    if torch.any(dist[..., 12] != 0) or torch.any(dist[..., 13] != 0):
+        invTilt = inverseTiltProjection(dist[..., 12], dist[..., 13])
+
+        # Transposed untilt points (instead of [x,y,1]^T, we obtain [x,y,1])
+        pointsUntilt = torch.stack([x, y, torch.ones(x.shape, dtype=x.dtype)], -1) @ invTilt.transpose(-2, -1)
```
Comment (edgarriba) [inline kornia/geometry/calibration/undistort.py:82]:
```suggestion
        pointsUntilt = torch.stack([x, y, torch.ones(x.shape, device=x.device, dtype=x.dtype)], -1) @ invTilt.transpose(-2, -1)
```
